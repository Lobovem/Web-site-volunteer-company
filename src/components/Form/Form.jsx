import photo from '../../img/form-photo.jpg';
import sending from '../../img/form-sending.svg';
import { Btn } from '../kit/Btn/Btn';
import { useState } from 'react';
import s from './Form.module.scss';
import { Link } from 'react-router-dom';

export const Form = () => {
  const [sendState, setSendState] = useState(false);
  const [inputDataForm, setInputDataForm] = useState({ name: '', email: '', textarea: '' });
  const [dataForm, setDataForm] = useState([]);

  const handleChangeState = () => {
    setSendState(!sendState);
    setTimeout(setSendState, 3000);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    handleChangeState();
    setDataForm([...dataForm, inputDataForm]);
    setInputDataForm({ name: '', email: '', textarea: '' });
  };

  const handleChange = (e) => {
    setInputDataForm({ ...inputDataForm, [e.target.name]: e.target.value });
  };

  return (
    <div className={s.form}>
      <div className={s.form__wrap}>
        <div className={s.form__imgWrap}>
          <img className={s.form__img} src={photo} alt="" />
        </div>

        {sendState ? (
          <div className={s.form__sending}>
            <img className={s.form__sendingImg} src={sending} alt="form-sending.svg" />
            <p className={s.form__sendingTitle}>
              <span>вітаємо!</span> дані успішно відправлені!
            </p>
          </div>
        ) : (
          <form className={s.form__inner} onSubmit={onSubmit} method="post" action="/">
            <h2 className={s.form__title}>зв'яжіться з нами</h2>
            <p className={s.form__desc}>Форма зворотнього зв'язку</p>

            <label className={s.form__inputTitle} htmlFor="name">
              Імʼя
            </label>
            <input className={s.form__input} type="text" name="name" id="name" required value={inputDataForm.name} onChange={handleChange} />

            <label className={s.form__inputTitle} htmlFor="email">
              Email
            </label>
            <input
              className={s.form__input}
              type="email"
              name="email"
              id="email"
              required
              pattern="^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$"
              value={inputDataForm.email}
              onChange={handleChange}
            />

            <label className={s.form__inputTitle} htmlFor="area">
              Коментар
            </label>
            <textarea className={s.form__textArea} name="textarea" id="area" required value={inputDataForm.textarea} onChange={handleChange}></textarea>

            <div className={s.form__btnWrap}>
              <p className={s.form__text}>
                Відправляючи форму, я приймаю{' '}
                <Link to="/user-condition" className={s.form__link}>
                  умови угоди користувача
                </Link>
              </p>
              <Btn title="надіслати" className="btn" type="submit" />
            </div>
          </form>
        )}
      </div>
    </div>
  );
};
