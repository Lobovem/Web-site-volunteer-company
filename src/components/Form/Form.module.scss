@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.form {
  background: $color-grey;

  &__wrap {
    @include layoutMain;

    @media(min-width: 750px) {
      display: grid;
      grid-template-columns: repeat(2, 1fr);
      gap: 20px;
    }

    @include lg-min {
      gap: 50px;
    }
  }


  &__img {
    width: 100%;
    background: $color-grey;
    border-radius: 15px;
    margin-bottom: 30px;

    @media(min-width: 750px) {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }

  &__inner {
    background: $color-white;
    display: flex;
    flex-direction: column;
    align-items: stretch;
    padding: 40px 50px;
    box-shadow: 0px 4px 15px rgba(2, 95, 167, 0.08);
    border-radius: 15px;

    @media(min-width: 750px) {
      padding: 40px 30px;
    }
  }

  &__title {
    font-weight: $font-weight-500;
    font-size: $font-size-20;
    line-height: 20px;
    text-align: center;
    text-transform: uppercase;
    color: $color-main;
    background: $gradient-main;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    margin: 0;

    @include lg-min {
      font-size: $font-size-38;
      line-height: 38px;
    }
  }

  &__desc {
    font-size: $font-size-14;
    line-height: 17px;
    text-align: center;
    margin-bottom: 40px;

    @include lg-min {
      font-size: $font-size-24;
      line-height: 24px;
    }

    @media (min-width: 1200px) {

      margin-bottom: 80px;
    }
  }

  &__inputTitle {
    font-size: $font-size-14;
    line-height: 12px;
    margin-bottom: 10px;
    margin-left: 20px;
  }

  &__input {
    margin-bottom: 10px;
    border: 2px solid $color-main;
    border-radius: 5px;
    height: 35px;
    padding: 0 10px;

    @include lg-min {
      margin-bottom: 30px;
    }

    @media (min-width: 1200px) {
      margin-bottom: 40px;
    }
  }

  &__textArea {
    margin-bottom: 20px;
    border: 2px solid $color-main;
    border-radius: 5px;
    padding: 10px;
    min-height: 100px;

    @include lg-min {
      margin-bottom: 40px;
    }


    @media (min-width: 1200px) {
      margin-bottom: 80px;
    }
  }

  &__btnWrap {
    margin: 0 auto;
  }
}