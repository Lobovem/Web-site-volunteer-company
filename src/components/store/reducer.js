import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  burgerMenuState: false,
};

export const appReducer = createSlice({
  name: 'app',
  initialState: initialState,
  reducers: {
    isOpenBurgerMenu: (state, action) => {
      state.burgerMenuState = !state.burgerMenuState;
    },
  },
});

export const burgerMenuSelector = (state) => state.app.burgerMenuState;

export const { isOpenBurgerMenu } = appReducer.actions;
export default appReducer.reducer;
