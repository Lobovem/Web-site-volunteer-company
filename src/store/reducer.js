import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  todoListState: [],
};

export const appReducer = createSlice({
  name: 'todoList',
  initialState: initialState,
  reducers: {
    // add_todo: (state, action) => {
    //   state.todoListState = [
    //     ...state.todoListState,
    //     {
    //       id: Math.random(),
    //       task: action.payload,
    //       complete: false,
    //       isEditing: false,
    //     },
    //   ];
    // },
    // delete_todo: (state, action) => {
    //   state.todoListState = state.todoListState.filter((todo) => todo.id !== action.payload.id);
    // },
  },
});

// export const userInputSelector = (state) => state.todoList.inputState;
// export const todoListSelector = (state) => state.todoList.todoListState;

// export const { add_todo } = todoReducer.actions;
export default appReducer.reducer;
