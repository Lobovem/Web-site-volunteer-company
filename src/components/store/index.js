// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { appReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: appReducer.reducer,
  },
});
