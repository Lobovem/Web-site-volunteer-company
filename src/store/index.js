// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { todoReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: todoReducer.reducer,
  },
});
