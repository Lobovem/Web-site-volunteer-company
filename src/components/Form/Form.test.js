import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Form } from './Form';
import { Provider } from 'react-redux';
import { store } from '../../redux/config/store';

describe('Form` component', () => {
  test('renders Form component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Provider store={store}>
        <Router>
          <Form />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('should render the title (H2) component form', () => {
    render(
      <Provider store={store}>
        <Router>
          <Form />
        </Router>
      </Provider>
    );
    const titleSection = screen.getByRole('heading', { level: 2 });
    expect(titleSection).toBeInTheDocument();
  });

  test('should render the title component form', () => {
    render(
      <Provider store={store}>
        <Router>
          <Form />
        </Router>
      </Provider>
    );
    const titleSection = screen.getByRole('heading', { name: /ЗВ'ЯЖІТЬСЯ З НАМИ/i });
    expect(titleSection).toBeInTheDocument();
  });

  test('renders and search Form component with 1 images', () => {
    render(
      <Provider store={store}>
        <Router>
          <Form />
        </Router>
      </Provider>
    );
    const images = screen.getAllByRole('img');
    expect(images).toHaveLength(1);
  });

  test('renders and search component button', () => {
    render(
      <Provider store={store}>
        <Router>
          <Form />
        </Router>
      </Provider>
    );
    const buttonElement = screen.getByRole('button', { type: 'button' });
    expect(buttonElement).toBeInTheDocument();
  });

  test('renders and search input and textArea of form', () => {
    render(
      <Provider store={store}>
        <Router>
          <Form />
        </Router>
      </Provider>
    );
    const buttonElement = screen.getAllByRole('textbox');
    expect(buttonElement).toHaveLength(3);
  });
});
