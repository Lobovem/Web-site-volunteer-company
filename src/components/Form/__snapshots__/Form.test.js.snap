// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form\` component renders Form component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form"
      >
        <div
          class="form__wrap"
        >
          <div
            class="form__imgWrap"
          >
            <img
              alt=""
              class="form__img"
              src="./img/form-photo.jpg"
            />
          </div>
          <form
            class="form__inner"
          >
            <h2
              class="form__title"
            >
              зв'яжіться з нами
            </h2>
            <p
              class="form__desc"
            >
              Форма зворотнього зв'язку
            </p>
            <label
              class="form__inputTitle"
              for="name"
            >
              Імʼя
            </label>
            <input
              class="form__input"
              id="name"
              name="name"
              required=""
              type="text"
              value=""
            />
            <label
              class="form__inputTitle"
              for="email"
            >
              Email
            </label>
            <input
              class="form__input"
              id="email"
              name="email"
              pattern="^[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
              required=""
              type="email"
              value=""
            />
            <label
              class="form__inputTitle"
              for="area"
            >
              Коментар
            </label>
            <textarea
              class="form__textArea"
              id="textarea"
              name="textarea"
              required=""
            />
            <div
              class="form__btnWrap"
            >
              <p
                class="form__text"
              >
                Відправляючи форму, я приймаю
                <a
                  class="form__link"
                  href="/user-condition"
                >
                  умови угоди користувача
                </a>
              </p>
              <button
                class="btn"
                type="submit"
              >
                надіслати
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form"
    >
      <div
        class="form__wrap"
      >
        <div
          class="form__imgWrap"
        >
          <img
            alt=""
            class="form__img"
            src="./img/form-photo.jpg"
          />
        </div>
        <form
          class="form__inner"
        >
          <h2
            class="form__title"
          >
            зв'яжіться з нами
          </h2>
          <p
            class="form__desc"
          >
            Форма зворотнього зв'язку
          </p>
          <label
            class="form__inputTitle"
            for="name"
          >
            Імʼя
          </label>
          <input
            class="form__input"
            id="name"
            name="name"
            required=""
            type="text"
            value=""
          />
          <label
            class="form__inputTitle"
            for="email"
          >
            Email
          </label>
          <input
            class="form__input"
            id="email"
            name="email"
            pattern="^[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
            required=""
            type="email"
            value=""
          />
          <label
            class="form__inputTitle"
            for="area"
          >
            Коментар
          </label>
          <textarea
            class="form__textArea"
            id="textarea"
            name="textarea"
            required=""
          />
          <div
            class="form__btnWrap"
          >
            <p
              class="form__text"
            >
              Відправляючи форму, я приймаю
              <a
                class="form__link"
                href="/user-condition"
              >
                умови угоди користувача
              </a>
            </p>
            <button
              class="btn"
              type="submit"
            >
              надіслати
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
